### PR TITLE
Basic overwriting update command for PodClusters

### PR DESCRIPTION
**Pull request by rudle**
I'm scared to add any more complexity than this to PC mutation - I'd
prefer to merge the JSON by hand using a good tool for the job (ie. a
scripting language).

### CODE REVIEW

**Comment by rudle** [comment]
PTAL. 
I've decided that I'm happy with things as they are - kingpin's required option parsing reduces the potential for a large class of errors. (ie. It's impossible to erase all annotations by forgetting to pass a command-line option)

File: bin/p2-pcctl/main.go [hunk 1]
```diff
@@ -39,10 +40,16 @@ var (
 	getAZ    = cmdGet.Flag("az", "The availability zone of the pod cluster").Required().String()
 	getName  = cmdGet.Flag("name", "The cluster name (ie. staging, production)").Required().String()
 
-	cmdDelete   = kingpin.Command(CmdDelete, "Show a pod cluster. ")
+	cmdDelete   = kingpin.Command(CmdDelete, "Delete a pod cluster. ")
 	deletePodID = cmdDelete.Flag("pod", "The pod ID on the pod cluster").Required().String()
 	deleteAZ    = cmdDelete.Flag("az", "The availability zone of the pod cluster").Required().String()
 	deleteName  = cmdDelete.Flag("name", "The cluster name (ie. staging, production)").Required().String()
+
+	cmdUpdate         = kingpin.Command(CmdUpdate, "Delete a pod cluster. ")
```
Comment (anthonybishopric) [inline bin/p2-pcctl/main.go]: s/Delete/Update

File: bin/p2-pcctl/main.go [hunk 2]
```diff
@@ -39,10 +40,16 @@ var (
...
+	updatePodID       = cmdUpdate.Flag("pod", "The pod ID on the pod cluster").Required().String()
+	updateAZ          = cmdUpdate.Flag("az", "The availability zone of the pod cluster").Required().String()
+	updateName        = cmdUpdate.Flag("name", "The cluster name (ie. staging, production)").Required().String()
+	updateAnnotations = cmdUpdate.Flag("annotations", "JSON string representing the complete update ").Required().String()
```
Comment (anthonybishopric) [inline bin/p2-pcctl/main.go]: 🚲  🏠  time - It'd be nice to have the ability to delete by ID at some point.

File: bin/p2-pcctl/main.go [hunk 3]
```diff
@@ -100,6 +107,30 @@ func main() {
...
+	case CmdUpdate:
+		az := fields.AvailabilityZone(*updateAZ)
+		cn := fields.ClusterName(*updateName)
+		podID := types.PodID(*updatePodID)
+		selector := selectorFrom(az, cn, podID)
+		pccontrol := control.NewPodCluster(az, cn, podID, pcstore, selector, kv.NewUnmanagedSession("pcctl", currentUserName()))
+		var annotations fields.Annotations
+		err := json.Unmarshal([]byte(*updateAnnotations), &annotations)
+		if err != nil {
+			_, _ = os.Stderr.Write([]byte(fmt.Sprintf("Annotations are invalid JSON. Err follows:\n%v", err)))
```
Comment (anthonybishopric) [inline bin/p2-pcctl/main.go]: Does `log.Fatalf` write to stdout or something?

**Comment by anthonybishopric** [comment]
👍  looks good, just nits

File: bin/p2-pcctl/main.go [hunk 4]
```diff
@@ -39,10 +40,16 @@ var (
...
+	cmdDelete   = kingpin.Command(CmdDelete, "Delete a pod cluster. ")
 	deletePodID = cmdDelete.Flag("pod", "The pod ID on the pod cluster").Required().String()
 	deleteAZ    = cmdDelete.Flag("az", "The availability zone of the pod cluster").Required().String()
 	deleteName  = cmdDelete.Flag("name", "The cluster name (ie. staging, production)").Required().String()
+
+	cmdUpdate         = kingpin.Command(CmdUpdate, "Update a pod cluster. ")
+	updatePodID       = cmdUpdate.Flag("pod", "The pod ID on the pod cluster").Required().String()
+	updateAZ          = cmdUpdate.Flag("az", "The availability zone of the pod cluster").Required().String()
+	updateName        = cmdUpdate.Flag("name", "The cluster name (ie. staging, production)").Required().String()
+	updateAnnotations = cmdUpdate.Flag("annotations", "JSON string representing the complete update ").Required().String()
```
Comment (petertseng) [inline bin/p2-pcctl/main.go]: oh noes extra space at the end

**Pull request merged**